src/main/java/org/powerflows/dmn/engine/model/decision/Decision.java

```diff
@@ -72,70 +73,97 @@ public ExpressionType getExpressionType() {
         return rules;
     }
 
-    public static <P extends AbstractBuilder> Builder<P> builder() {
-        return new Builder<>();
+    public static FluentBuilder fluentBuilder() {
+        return new FluentBuilder();
     }
 
-    public static final class Builder<P extends AbstractBuilder> extends AbstractBuilder<Decision> {
-
-        private Builder() {
-        }
+    public static Builder builder() {
+        return new Builder();
+    }
 
+    public static abstract class DecisionBuilder<B extends DecisionBuilder<B>> extends AbstractBuilder<Decision> {
         @Override
         protected void initProduct() {
             this.product = new Decision();
         }
 
-        public Builder<P> id(String id) {
+        public B id(String id) {
             this.product.id = id;
 
-            return this;
+            return (B) this;
```

Comment:
here will be unchecked cast

Reply:
true, but there is no other way, the only implementation is here so the cast is always valid